### PR TITLE
Fix use locks

### LINKED ISSUE #122: if volplugin map/mount fails, rbd lock is not freed in some situations

### PR DESCRIPTION
**Pull request by erikh**
resolves #122 #121. Incorporates #120 so please review that first.

right now this is nearly impossible to test, it literally has to have a broken ceph to happen.

/cc @mapuri @unclejack

### CODE REVIEW

File: storage/backend/ceph/ceph.go [hunk 1]
```diff
@@ -82,16 +82,20 @@ func (c *Driver) unlockAndLog(do storage.DriverOptions) {
 // Format formats a created volume.
 func (c *Driver) Format(do storage.DriverOptions) error {
 	device, err := c.mapImage(do)
-	if err != nil {
+	if err == ErrLockFailed {
```
Comment (mapuri) [inline storage/backend/ceph/ceph.go]: looks like we try to unlock as part of error handling so wondering, if this check should be `err != ErrLockFailed`?

I am also wondering why can't we unlock always whenever `err != nil`, that way we can just `defer` the call to unlock when err is not nil?

Comment (erikh) [inline storage/backend/ceph/ceph.go]: let me look at this patch again. I don't think it's very sound and I created it in a rush.

**Comment by erikh** [comment]
closing this; going to overhaul some of the code in here instead.

**Comment by erikh** [comment]
thanks for catching my sloppiness in review; i'm not sure how I thought this was a good idea.

**Pull request closed without merging**